akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -1250,7 +1259,7 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
 
   def receiveGetReplicaCount(): Unit = {
     // selfAddress is not included in the set
-    sender() ! ReplicaCount(nodes.size + 1)
+    sender() ! ReplicaCount(nodes.size + weaklyUpNodes.size + 1)
```

Comment:
We need to decide. I'm on the fence but think the safest approach is to only include the WeaklyUp in the gossip and not include them in the direct replication. That is what you have implemented apart from that then the WeaklyUp should not be included in this count.